Source/CarthageKit/Project.swift

```diff
@@ -187,11 +187,19 @@ public final class Project {
 		}
 
 		return cartfile.zipWith(privateCartfile)
-			.map { (var cartfile, privateCartfile) -> Cartfile in
-				cartfile.appendCartfile(privateCartfile)
+			.map { (var cartfile, privateCartfile) -> ColdSignal<Cartfile> in
+				// TODO: use Set when Carthage supports Swift 1.2
+				let duplicateDeps = filter(privateCartfile.dependencies) { privateDep in
+					contains(cartfile.dependencies, {$0.project == privateDep.project})
+				}
```

Comment:
Those functions could go into [`FrameworkExtensions.swift`](https://github.com/Carthage/Carthage/blob/1ee40ca0dc982abbf88b79f35cefd5e67fa62198/Source/CarthageKit/FrameworkExtensions.swift).

> Tentative output:

Could we make it clearer that the `Cartfile` part of the line is the source? Maybe something like:

```
github "dup1/dup1" (found in Cartfile and Cartfile.private)
```